codex-rs/core/src/config_loader/mod.rs

```diff
@@ -0,0 +1,335 @@
+#[cfg(target_os = "macos")]
+mod macos;
+
+#[cfg(target_os = "macos")]
+use std::any::Any;
+use std::fs;
+use std::future::Future;
+use std::io;
+use std::path::Path;
+use std::path::PathBuf;
+use tokio::task;
+use toml::Value as TomlValue;
+
+const CONFIG_TOML_FILE: &str = "config.toml";
+const MANAGED_CONFIG_TOML_FILE: &str = "managed_config.toml";
+
+#[derive(Debug)]
+pub(crate) struct LoadedConfigLayers {
+    pub base: TomlValue,
+    pub managed_config: Option<TomlValue>,
+    pub managed_preferences: Option<TomlValue>,
+}
+
+// Configuration layering pipeline (top overrides bottom):
+//
+//        +-------------------------+
+//        | Managed preferences (*) |
+//        +-------------------------+
+//                    ^
+//                    |
+//        +-------------------------+
+//        |  managed_config.toml   |
+//        +-------------------------+
+//                    ^
+//                    |
+//        +-------------------------+
+//        |    config.toml (base)   |
+//        +-------------------------+
+//
+// (*) Only available on macOS via managed device profiles.
+
+pub async fn load_config_as_toml_async(codex_home: PathBuf) -> io::Result<TomlValue> {
+    let LoadedConfigLayers {
+        mut base,
+        managed_config,
+        managed_preferences,
+    } = load_config_layers_async(codex_home).await?;
+
+    for overlay in [managed_config, managed_preferences].into_iter().flatten() {
+        merge_toml_values(&mut base, &overlay);
+    }
+
+    Ok(base)
+}
+
+pub fn load_config_as_toml(codex_home: &Path) -> io::Result<TomlValue> {
+    block_on_config_future(load_config_as_toml_async(codex_home.to_path_buf()))
+}
+
+pub(crate) async fn load_config_layers_async(
+    codex_home: PathBuf,
+) -> io::Result<LoadedConfigLayers> {
+    let user_config = read_config_from_path_async(codex_home.join(CONFIG_TOML_FILE), true).await?;
+    let managed_config =
+        read_config_from_path_async(codex_home.join(MANAGED_CONFIG_TOML_FILE), false).await?;
+    let managed_preferences = load_managed_admin_config_layer_async().await?;
+
+    Ok(LoadedConfigLayers {
+        base: user_config.unwrap_or_else(default_empty_table),
+        managed_config,
+        managed_preferences,
+    })
+}
+
+pub(crate) fn load_config_layers(codex_home: &Path) -> io::Result<LoadedConfigLayers> {
+    block_on_config_future(load_config_layers_async(codex_home.to_path_buf()))
+}
+
+fn default_empty_table() -> TomlValue {
+    TomlValue::Table(Default::default())
+}
+
+async fn read_config_from_path_async(
```

Comment:
In general, there is no need to add the `_async` suffix to a method. I would only do it when there are both sync and non-sync versions. In this case, I think we only need the async version, for example?